### PR TITLE
Fix documentation for writing Molden files for CI NO roots

### LINKED ISSUE #1638: Molden output for CI not correctly documented
I overlooked this when previously checking the documentation. The CI/MCSCF example from the following page does not work: http://www.psicode.org/psi4manual/1.3/molden.html. A minor issue is that `'ci'` is not a valid method. The second problem is that the `wfn.opdm` method should be replaced by `wfn.get_opdm`. (A useful example can be found in the source tree: `tests/molden2/input.dat`).

I believe this is a correct example, but it would be nice if someone more familiar with the CI code could confirm and explain a few questions below. (I would be happy to make a PR with updated documentation.)

```
molecule {
0 1
H 0.0 0.0 0.0
H 0.0 0.0 1.0
}

set basis cc-pVTZ
set opdm true

E, wfn = energy('fci', return_wfn=True)
molden(wfn, 'no_root1.molden', density_a=wfn.get_opdm(0, 0, "A", True))
```

Main questions:
- Is this example correct? It runs and seems to give sensible results.
- Which levels of theory (other than FCI and CASSCF) can be used? (CISD gives an error: `AttributeError: 'psi4.core.Wavefunction' object has no attribute 'get_opdm'`.)
- How can one use the first two arguments to `get_opdm`? They are documented here in the source code: https://github.com/psi4/psi4/blob/880f87f5b2765ed00efaf6dd63c406b574c40157/psi4/src/psi4/detci/ciwave.h#L141 but setting anything else than `0, 0` for the first two arguments does not work: `Fatal Error: CIWavefunction::get_opdm: Requested OPDM was not formed!` I tried to add `set tdm true`, but this had no effect. (Also, I guess writing out a transition density matrix to a Molden file makes little sense.)
- When the last argument is `True`, does this mean the frozen-core is included in the density matrix?

### PR DESCRIPTION
**Pull request by tovrstra**
Fixes #1638 

## Description
Corrects some small errors in the documentation of the molden writer, plus improved formatting.

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

File: psi4/driver/driver.py [hunk 1]
```diff
@@ -1740,24 +1740,34 @@ def molden(wfn, filename=None, density_a=None, density_b=None, dovirtual=None):
...
+       ``fci``, ``casscf``, etc. The first two arguments of ``get_opdm`` can be
+       set to ``n, n`` with n > 0 to write densities for higher roots, provided
+       these roots were computed, see :term:`NUM_ROOTS <NUM_ROOTS (DETCI)>`. The
+       third argument controls the spin (``A``, ``B`` or ``SUM``) and the final
+       boolean option determines whether inactive orbitals are included.
+
+       >>> E, wfn = energy('detci', return_wfn=True)
+       >>> molden(wfn, 'no_root1.molden', density_a=wfn.get_opdm(0, 0, "A", True))
+
+    3. The following **does NOT work**, please see below.
```
Comment (PeterKraus) [inline psi4/driver/driver.py]: Does it not work (crash) or does it just print incorrect values due to the wrong basis? Perhaps add a comment here, saying something on the lines of "molden spec requires densities in a MO basis"

File: psi4/driver/driver.py [hunk 2]
```diff
@@ -1740,24 +1740,34 @@ def molden(wfn, filename=None, density_a=None, density_b=None, dovirtual=None):
...
-    >>> molden(wfn, 'ccsd_no.molden', density_a=Da_mo)
+    1. Molden file for DFT calculation.
+
+       >>> E, wfn = energy('b3lyp', return_wfn=True)
+       >>> molden(wfn, 'mycalc.molden')
+
+    2. Molden file for CI/MCSCF computation using NO roots.
+       Any method returning a ``CIWavefunction`` object will work: ``detci``,
+       ``fci``, ``casscf``, etc. The first two arguments of ``get_opdm`` can be
+       set to ``n, n`` with n > 0 to write densities for higher roots, provided
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply psi4/driver/driver.py]: ` n >= 0` where `(0, 0)` is the ground state...?

File: psi4/driver/driver.py (hunk 1 above)
Comment (tovrstra) [reply psi4/driver/driver.py]: Done

File: psi4/driver/driver.py [hunk 3]
```diff
@@ -1740,24 +1740,34 @@ def molden(wfn, filename=None, density_a=None, density_b=None, dovirtual=None):
...
+       these roots were computed, see :term:`NUM_ROOTS <NUM_ROOTS (DETCI)>`. The
+       third argument controls the spin (``A``, ``B`` or ``SUM``) and the final
```
[the remark this answers is not in the document]
Comment (tovrstra) [reply psi4/driver/driver.py]: Done

File: psi4/driver/driver.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tovrstra) [reply psi4/driver/driver.py]: Done

**Comment by tovrstra** [comment]
@PeterKraus I've tried to clarify everything, also some improvements in the existing texts. I've also removed the `properties=['dipole']` argument because it is no longer needed.

**Pull request merged**